config/config_env.go

```diff
@@ -159,46 +159,37 @@ func reporterConfigFromEnv() (*ReporterConfig, error) {
 		}
 	}
 
-	host := jaeger.DefaultUDPSpanServerHost
-	ep := os.Getenv(envEndpoint)
-	if e := os.Getenv(envAgentHost); e != "" {
-		if ep != "" {
-			return nil, errors.Errorf("cannot set env vars %s and %s together", envAgentHost, envEndpoint)
+	if e := os.Getenv(envEndpoint); e != "" {
+		u, err := url.ParseRequestURI(e)
+		if err != nil {
+			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envEndpoint, e)
 		}
-		host = e
-	}
-
-	port := jaeger.DefaultUDPSpanServerPort
-	if e := os.Getenv(envAgentPort); e != "" {
-		if ep != "" {
-			return nil, errors.Errorf("cannot set env vars %s and %s together", envAgentPort, envEndpoint)
+		rc.CollectorEndpoint = u.String()
+		user := os.Getenv(envUser)
+		pswd := os.Getenv(envPassword)
+		if user != "" && pswd == "" || user == "" && pswd != "" {
+			return nil, errors.Errorf("you must set %s and %s env vars together", envUser, envPassword)
 		}
-		if value, err := strconv.ParseInt(e, 10, 0); err == nil {
-			port = int(value)
-		} else {
-			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envAgentPort, e)
+		rc.User = user
+		rc.Password = pswd
+	} else {
+		host := jaeger.DefaultUDPSpanServerHost
+		if e := os.Getenv(envAgentHost); e != "" {
+			host = e
 		}
-	}
-
-	// the side effect of this is that we are building the default value, even if none of the env vars
-	// were not explicitly passed
-	rc.LocalAgentHostPort = fmt.Sprintf("%s:%d", host, port)
 
-	if ep != "" {
-		u, err := url.ParseRequestURI(ep)
-		if err != nil {
-			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envEndpoint, ep)
+		port := jaeger.DefaultUDPSpanServerPort
+		if e := os.Getenv(envAgentPort); e != "" {
+			if value, err := strconv.ParseInt(e, 10, 0); err == nil {
+				port = int(value)
+			} else {
+				return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envAgentPort, e)
+			}
 		}
-		rc.CollectorEndpoint = fmt.Sprintf("%s", u)
-	}
-
-	user := os.Getenv(envUser)
-	pswd := os.Getenv(envPassword)
-	if user != "" && pswd == "" || user == "" && pswd != "" {
-		return nil, errors.Errorf("you must set %s and %s env vars together", envUser, envPassword)
+		// the side effect of this is that we are building the default value, even if none of the env vars
+		// were explicitly passed
```

Comment:
let's remove this comment, II don't know what it is trying to say, only confuses people

Reply:
Ok let me remove.
It is there by original author.